Documentation/botr/readytorun-format.md

```diff
@@ -206,6 +206,7 @@ fixup kind, the rest of the signature varies based on the fixup kind.
 | READYTORUN_FIXUP_DelegateCtor            |  0x2C | Delegate constructor, followed by method signature
 | READYTORUN_FIXUP_DeclaringTypeHandle     |  0x2D | Dictionary lookup for method declaring type. Followed by the type signature.
 | READYTORUN_FIXUP_IndirectPInvokeTarget   |  0x2E | Target of an inlined PInvoke. Followed by method signature.
+| READYTORUN_FIXUP_IndirectPInvokeTargetSuppressGC |  0x2F | Target of an inlined PInvoke (suppress GC). Followed by method signature.
```

Comment:
This should be just `READYTORUN_FIXUP_PInvokeTarget`.

- There is nothing SuppressGC-specific about it
- It should have one less indirection compared to regular PInvokeTarget

Reply:
> There is nothing SuppressGC-specific about it

We use the SuppressGC attribute to determine that level of indirection is possible so it is related. Clearly not required, but that is the metric at the moment. I am all for changing this but it does seem to obscure the intent of the fixup.

Reply:
The difference between `IndirectPInvokeTarget` and `PInvokeTarget` is in when they are resolved. `IndirectPInvokeTarget` is resolved right before the PInvoke call. `PInvokeTarget` is resolved before the containing method starts executing.

We have decided to couple the laziness and SupressGC behaviors together in the JIT/compiler, but they do not need to be coupled in the R2R file format.

Reply:
I think I am missing some nuance here. The impetus for adding this new field was based on feedback about the cost of querying for a custom attribute. Along with adding that an additional feature would be to remove a level of indirection. I chose not to do that because it required more code change with arguable wasn't that impactful. Removing a layer of indirection is straight forward so I can do that. However, the original issue was perf due to the semantics of how it was implemented. Removing any mention of that now would obscure that when in fact this fixup type can be much more expensive due to the custom attribute lookup. 

I think we are obscuring a semantic of this lookup that was important enough to address and so should be conveyed in any new name. I am find with dropping `Suppress_GC`, but would then suggest it become `PInvokeTarget_Slow` or some other indication that it is non-optimal so when we have the real option it can use the name `PInvokeTarget`.

Reply:
> the original issue was perf due to the semantics

The root cause of this issue is that IndirectPInvokeTarget cannot be used with suppress GC because of it triggers GC in a no-trigger region.

All other combinations work fine:
- SupressGC + PInvokeTarget
- No SupressGC + PInvokeTarget (this combination is not used anywhere today, but it may be useful one day)
- No SupressGC + IndirectPInvokeTarget

> Removing any mention of that now would obscure that when in fact this fixup type can be more expensive due to the custom attribute lookup.

You can add comment about this.